agent/lib_aws_sdk_php.c

```diff
@@ -14,10 +14,402 @@
+   */
+  message_params->destination_name = queue_name;
+  cloud_attrs->cloud_account_id = account_id;
+  cloud_attrs->cloud_region = region;
+}
+
+char* nr_lib_aws_sdk_php_get_command_arg_value(char* command_arg_name,
+                                               NR_EXECUTE_PROTO) {
+  zval* param_array = NULL;
+  zval* command_arg_array = NULL;
+  char* command_arg_value = NULL;
+
+  if (NULL == command_arg_name) {
+    return NULL;
+  }
+  /* To extract the Aws/AwsClient::__call $argument, we get the second arg. */
+  param_array = nr_php_arg_get(2, NR_EXECUTE_ORIG_ARGS);
+
+  if (nr_php_is_zval_valid_array(param_array)) {
+    /* The first element in param_array is an array of parameters. */
+    command_arg_array = nr_php_zend_hash_index_find(Z_ARRVAL_P(param_array), 0);
+    if (nr_php_is_zval_valid_array(command_arg_array)) {
+      zval* queueurl_arg = nr_php_zend_hash_find(Z_ARRVAL_P(command_arg_array),
+                                                 command_arg_name);
+
+      if (nr_php_is_zval_non_empty_string(queueurl_arg)) {
+        command_arg_value = nr_strdup(Z_STRVAL_P(queueurl_arg));
+      }
+    }
+  }
+
+  nr_php_arg_release(&param_array);
+  return command_arg_value;
+}
+
+/*
+ * For Aws/AwsClient::__call see
+ * https://github.com/aws/aws-sdk-php/blob/master/src/AwsClientInterface.php
+ * ALL
+ * client commands are handled by this function, so it is the start and end of
+ * any command. Creates and executes a command for an operation by name.
+ * When a class command isn't explicitly created as a function, the __call class
+ * handles the invocation.  This means all AWS Client Service commands are
+ * handled by this call. Any invocation starts when this function starts, and
+ * ends when it ends.  This function decodes the command name, determines the
+ * appropriate args, decodes the args, generates a guzzle request to send to the
+ * AWS Service, gets the guzzle response from the AWS Service, and bundles that
+ * response into an AswResult to return.
+ *
+ * @param string $name      Name of the command to execute.
+ * @param array  $arguments Arguments to pass to the getCommand method.
+ *
+ * @return ResultInterface
+ * @throws \Exception
+ */
+
+NR_PHP_WRAPPER(nr_aws_client_call_before) {
+  (void)wraprec;
+  nr_segment_t* segment = NULL;
+  /*
+   * Start a segment in case it needs to become an external, message, or
+   * datastore segment.
+   */
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+  if (NULL != segment) {
+    segment->wraprec = auto_segment->wraprec;
+  }
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_aws_client_call) {
+  (void)wraprec;
+
+  zval* command_name = NULL;
+  const char* klass = NULL;
+  char* command_name_string = NULL;
+  char* real_class_and_command = NULL;
+  nr_segment_t* segment = NULL;
+  zend_class_entry* class_entry = NULL;
+  int klass_len = 0;
+
+  class_entry = Z_OBJCE_P(nr_php_execute_scope(execute_data));
+  klass = nr_php_class_entry_name(class_entry);
```

Comment:
Add a check if `NULL != class_entry`:
```suggestion
  if (NULL == class_entry) {
    return;
  }
  klass = nr_php_class_entry_name(class_entry);
```

Reply:
Cannot do a simple `return;` inside a wrapper.
See this [comment](https://github.com/newrelic/newrelic-php-agent/pull/1003#discussion_r1915836283) for the link to understand what logic goes into the return value.

`goto end;` is the way to exit early from a wrapper.
[1523be8](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/1523be8657cb7c18c98f51cad3e65250005a2063)